core/client/fs/src/main/java/alluxio/client/file/CachingFileSystem.java

```diff
@@ -87,9 +87,10 @@ public URIStatus getStatus(AlluxioURI path, GetStatusPOptions options)
   public List<URIStatus> listStatus(AlluxioURI path, ListStatusPOptions options)
       throws FileDoesNotExistException, IOException, AlluxioException {
     checkUri(path);
-    List<URIStatus> statuses = super.listStatus(path, options);
-    for (URIStatus status : statuses) {
-      mMetadataCache.put(status.getPath(), status);
+    List<URIStatus> statuses = mMetadataCache.listStatus(path);
```

Comment:
as discussed offline, let us selectively cache `listStatus` result only on non-recursive queries.
@chcn please also comment the pros/cons here for future references.

Reply:
@apc999 updated, PTAL